modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/JavaUpgradeUpdateCommerceAddressCheck.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.source.formatter.check.util.SourceUtil;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class JavaUpgradeUpdateCommerceAddressCheck extends BaseJavaTermCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, JavaTerm javaTerm,
+			String fileContent)
+		throws Exception {
+
+		List<String> importNames = javaTerm.getImportNames();
+
+		String javaTermContent = javaTerm.getContent();
+
+		if (!importNames.contains(
+				"com.liferay.commerce.model.CommerceAddress")) {
+
+			return javaTermContent;
+		}
+
+		Matcher matcher = _pattern.matcher(javaTermContent);
+
+		if (!matcher.find() ||
+			!_isCommerceAddress(javaTermContent, fileContent, matcher)) {
+
+			return javaTermContent;
+		}
+
+		String line = matcher.group();
+
+		String newLine = StringUtil.replace(
+			line, matcher.group(1),
+			_getNewUpdateCommerceAddressImplementation(
+				SourceUtil.getIndent(line), matcher.group(2)));
+
+		return StringUtil.replace(javaTermContent, line, newLine);
+	}
+
+	@Override
+	protected String[] getCheckableJavaTermNames() {
+		return new String[] {JAVA_METHOD};
+	}
+
+	private String _getNewUpdateCommerceAddressImplementation(
+		String indent, String variable) {
+
+		StringBundler sb = new StringBundler(7);
+
+		sb.append(StringPool.OPEN_PARENTHESIS);
+
+		for (String method : _METHODS) {
+			sb.append(StringPool.NEW_LINE);
+			sb.append(indent);
+			sb.append(StringPool.TAB);
+			sb.append(variable);
+			sb.append(StringPool.PERIOD);
+			sb.append(method);
+			sb.append("(),");
+		}
+
+		sb.append(StringPool.NEW_LINE);
+		sb.append(indent);
+		sb.append(StringPool.TAB);
+		sb.append("ServiceContextThreadLocal.getServiceContext()");
```

Comment:
We need to make sure we import `ServiceContextThreadLocal` if replacements need to be made. Otherwise, the code will be missing an import and will not compile.